multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UploadAppStep.java

```diff
@@ -75,7 +75,10 @@ public StepPhase executeAsyncStep(ProcessContext context) throws FileStorageExce
 
         boolean contentChanged = detectApplicationFileDigestChanges(context, cloudApp, newApplicationDigest);
         if (contentChanged) {
-            return proceedWithUpload(context, applicationToProcess, moduleFileName);
+            proceedWithUpload(context, applicationToProcess, moduleFileName);
+            attemptToUpdateApplicationDigest(context.getControllerClient(), cloudApp, newApplicationDigest);
+            setAppContentChanged(context, true);
```

Comment:
Why not including setAppContentChanged in proceedWithUpload()?

Reply:
done